conjure-core/src/main/java/com/palantir/parsec/Parsers.java

```diff
@@ -108,28 +108,47 @@ public T parse(ParserState input) throws ParseException {
     /**
      * Parse either firstOption or iterate through otherOptions and return result.
      *
+     * Exceptions are caught while trying options - if no options succeed, then the last exception thrown is re-thrown
+     * as a {@code ParseError}.
+     *
      * @param <T> the type the parsers return
      * @param firstOption the first parser to try
      * @param otherOptions additional parsers to try
      * @return the specified parser
      */
     @SafeVarargs
     public static <T> Parser<T> or(final Parser<? extends T> firstOption, final Parser<? extends T>... otherOptions) {
-        return new Parser<T>() {
-            @Override
-            public T parse(ParserState input) throws ParseException {
-                T result = gingerly(firstOption).parse(input);
+        return input -> {
+            T result;
+            RuntimeException exception = null;
+
+            try {
+                result = gingerly(firstOption).parse(input);
+            } catch (RuntimeException e) {
+                exception = e;
+                result = null;
+            }
+
+            if (result != null) {
+                return result;
+            }
+            for (Parser<? extends T> nextOption : otherOptions) {
+                try {
+                    result = gingerly(nextOption).parse(input);
+                } catch (RuntimeException e) {
+                    exception = e;
```

Comment:
Maybe something along these lines
```suggestion
                    if (exception == null) {
                        exception = e;
                    } else {
                        exception.addSuppressed(e);
                    }
```

Reply:
Do we also need to `input.rewind();` in this case?

Reply:
Oh I wasn't aware of suppressed exceptions - looks great!

Reply:
What does rewinding achieve if we're throwing a `ParseException`?

Reply:
Nevermind - I see.